# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/loader/mozJSComponentLoader.cpp
# Commit: 6450088b6b73
# Full Hash: 6450088b6b73ffc17c79cd7097b1bfe30d00e207
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-09-14 21:29:41
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1656261 - Back out all recent StartupCache work r=RyanVM
#   
#   This backs out all work from bug 1627075 as well as all of its
#   descendents. There were a few conflicts when backing this out but
#   overall it was pretty clean, so I would say it's a fairly mild
# ==============================================================================

diff -r b0fabcd60430 -r 6450088b6b73 js/xpconnect/loader/mozJSComponentLoader.cpp
--- a/js/xpconnect/loader/mozJSComponentLoader.cpp	Mon Sep 14 17:32:40 2020 +0000
+++ b/js/xpconnect/loader/mozJSComponentLoader.cpp	Mon Sep 14 17:00:53 2020 +0000
@@ -22,7 +22,7 @@
 #include "js/CharacterEncoding.h"
 #include "js/CompilationAndEvaluation.h"
 #include "js/friend/JSMEnvironment.h"  // JS::ExecuteInJSMEnvironment, JS::GetJSMEnvironmentOfScriptedCaller, JS::NewJSMEnvironment
-#include "js/Object.h"  // JS::GetCompartment
+#include "js/Object.h"                 // JS::GetCompartment
 #include "js/Printf.h"
 #include "js/PropertySpec.h"
 #include "js/SourceText.h"  // JS::SourceText
@@ -54,7 +54,6 @@
 #include "mozilla/ClearOnShutdown.h"
 #include "mozilla/MacroForEach.h"
 #include "mozilla/Preferences.h"
-#include "mozilla/Omnijar.h"
 #include "mozilla/ResultExtensions.h"
 #include "mozilla/ScriptPreloader.h"
 #include "mozilla/ScopeExit.h"
@@ -665,10 +664,6 @@
 
 static mozilla::Result<nsCString, nsresult> ReadScript(
     ComponentLoaderInfo& aInfo) {
-  // We're going to cache the XDR encoded script data - suspend writes via the
-  // CacheAwareZipReader, otherwise we'll end up redundantly caching scripts.
-  AutoSuspendStartupCacheWrites suspendScache;
-
   MOZ_TRY(aInfo.EnsureScriptChannel());
 
   nsCOMPtr<nsIInputStream> scriptStream;
@@ -730,15 +725,7 @@
   // to loading the script, since we can always slow-load.
 
   bool writeToCache = false;
-
-  // Since we are intending to cache these buffers in the script preloader
-  // already, caching them in the StartupCache tends to be redundant. This
-  // ought to be addressed, but as in bug 1627075 we extended the
-  // StartupCache to be multi-process, we just didn't want to propagate
-  // this problem into yet more processes, so we pretend the StartupCache
-  // doesn't exist if we're not the parent process.
-  StartupCache* cache =
-      XRE_IsParentProcess() ? StartupCache::GetSingleton() : nullptr;
+  StartupCache* cache = StartupCache::GetSingleton();
 
   aInfo.EnsureResolvedURI();
 